messaging/src/main/java/org/axonframework/configuration/DefaultComponentRegistry.java

```diff
@@ -246,6 +260,12 @@ public DefaultComponentRegistry setOverridePolicy(@Nonnull OverridePolicy overri
 
     @Override
     public DefaultComponentRegistry disableEnhancer(Class<? extends ConfigurationEnhancer> enhancerClass) {
+        if (invokedEnhancers.contains(enhancerClass)) {
+            logger.warn("Disabling Configuration Enhancer [{}] won't take effect as it has already been invoked. "
```

Comment:
Nice! :) 
Sure you don't want this to be an exception though? :thinking: Configuration and components are taking effect that the user doesn't want.